src/systems/world_exporter/WorldExporter.cc

```diff
@@ -0,0 +1,331 @@
+          }
+          mesh = meshManager->Load(fullPath);
+
+          igndbg << "Submesh count : " << mesh->SubMeshCount() << std::endl;
+
+          if (!mesh) {
+            ignerr << "MESH NOT FOUND!" << std::endl;
+            return true;
+          }
+
+          for (unsigned int k = 0; k < mesh->SubMeshCount(); k++)
+          {
+            igndbg << ">>>>>>>>> SUBMESH : " << k << std::endl;
+            int j = mesh->SubMeshByIndex(k).lock()->MaterialIndex();
+
+            igndbg << "Query Material index : " << j << std::endl;
+
+            igndbg << "Material : " << mesh->MaterialByIndex(
+              mesh->SubMeshByIndex(k).lock()->MaterialIndex()) << std::endl;
+            igndbg << "Material Ptr: " << mesh->MaterialByIndex(
+              mesh->SubMeshByIndex(k).lock()->MaterialIndex()).get()
+              << std::endl;
+
+            igndbg << "Material index on worldmesh: " <<
+              worldMesh.IndexOfMaterial(mesh->MaterialByIndex(
+                mesh->SubMeshByIndex(k).lock()->MaterialIndex()).get())
+                << std::endl;
+
+            int i = 0;
+            if ( j != -1 )
+            {
+              i = worldMesh.IndexOfMaterial(mesh->MaterialByIndex(
+                mesh->SubMeshByIndex(k).lock()->MaterialIndex()).get());
+              if (i < 0)
+              {
+                i = worldMesh.AddMaterial(mesh->MaterialByIndex(
+                  mesh->SubMeshByIndex(k).lock()->MaterialIndex() ));
+              }
+            }
+            else
+            {
+              i = worldMesh.AddMaterial(mat);
+            }
+
+            igndbg << "Inserted Material index : " << i << std::endl;
+
+            igndbg << "Texture Image : "
+              << worldMesh.MaterialByIndex(i)->TextureImage()
+              << std::endl;
+
+            subm = worldMesh.AddSubMesh(
+              *mesh->SubMeshByIndex(k).lock().get());
+            subm.lock()->SetMaterialIndex(i);
+            igndbg << "Scale : " << _geom->Data().MeshShape()->Scale()
+              << std::endl;
+
+            subm.lock()->Scale(_geom->Data().MeshShape()->Scale());
+            subMeshMatrix.push_back(matrix);
+
+            igndbg << ">>>>>>>>>" << std::endl;
+          }
+      }
+      else
+      {
+        ignwarn << "Unsupported geometry type" << std::endl;
+      }
+
+      return true;
+    });
+
+    common::ColladaExporter exporter;
+    exporter.Export(&worldMesh, "./" + worldMesh.Name(), true,
+                    subMeshMatrix);
+    this->exported = true;
+  }
+};
+
+/////////////////////////////////////////////////
+WorldExporter::WorldExporter() : System(),
+    dataPtr(std::make_unique<WorldExporterPrivate>())
+{
+}
+
+/////////////////////////////////////////////////
+WorldExporter::~WorldExporter() = default;
+
+/////////////////////////////////////////////////
+void WorldExporter::PostUpdate(const UpdateInfo & /*_info*/,
+    const EntityComponentManager &_ecm)
+{
+  this->dataPtr->Export(_ecm);
```

Comment:
The fact that the world is exported immediately after the system is loaded might not work for the SubT use case. This is because the robots are loaded via the `/world/<world name>create_multiple` ign-transport service which would most likely happen after `WorldExporter::PostUpdate` is called the first time. My suggestion would be to create an ign-transport service in this system that triggers the export.

If dynamically loaded models need not be in the exported mesh, please ignore this comment.

Reply:
Dynamically loaded models are not needed for the subt use case.